### PR TITLE
Ctw 1356/request history last refresh

### PR DESCRIPTION
**Pull request by luthfib**
Jira Ticket Number: https://zeushealth.atlassian.net/browse/CTW-1356

## What does this PR accomplish?

Fixes a bug showed no requested history even though a patient had data. Essentially when patients had multiple patient records we at random chose one FHIR id and use that as the one we used to hit the patient history API. This behavior is incorrect as we may have pulled the data under one of the other patient id's. We now look at at all the fhir id's under a patient. This is accomplished by iterating over all the patient id's for a given patient. 

## How did you test it?

I tested this by navigating over to the patient in question mentioned in the bug report and verifying that we indeed now show a requested date.

### CODE REVIEW

**Review by WesleyKapow — COMMENTED** [review]
Mostly questions trying to understand what's going on.

Also, for the PR description can you:
1. Add jira ticket
2. Explain what the bug was that you have fixed. Why was it wrong before?

File: src/components/content/patient-history/use-patient-history.tsx [hunk 1]
```diff
@@ -142,40 +143,64 @@ export async function getBuilderRefreshHistoryMessages({
...
+
+    // Iterate over each responses and iterate over response data array to find the latest status out of all resposnes
+    const latestDones = responses.map((response) =>
+      find(response.data, (item) =>
+        item.attributes.providers.every((provider) => provider.status === "done")
+      )
+    );
+
+    // Get index of latest done to ensure we get correct matching status
+    const latestDoneIndex = latestDones.findIndex((latestDone) => latestDone !== undefined);
```
[the remark this answers is not in the document]
Comment (luthfib) [reply src/components/content/patient-history/use-patient-history.tsx:169]: Yup! The api is sorted that way.

File: src/fhir/patient-helper.ts [hunk 2]
```diff
@@ -2,19 +2,52 @@ import { SearchParams } from "fhir-kit-client";
...
+        lastUpdated: "DESC",
+      },
+    });
+    return data.PatientConnection.edges
+      .filter(
+        (p) =>
+          !p.node.meta?.tag?.some(
+            (t) => t.system === SYSTEM_ZUS_UPI_RECORD_TYPE && t.code === "universal"
+          )
+      )
```
[the remark this answers is not in the document]
Comment (luthfib) [reply src/fhir/patient-helper.ts:44]: We only want upid resources here, so essentially filtering for those. 

DISCLAIMER: I took this straight from this pr that is in draft https://github.com/zeus-health/ctw-component-library/pull/1359

[the remark this answers is not in the document]
Comment (WesleyKapow) [reply src/fhir/patient-helper.ts:44]: Ugh remind me, what are upid resources? Aren't we looking things up by upid?

File: src/components/content/patient-history/use-patient-history.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (WesleyKapow) [reply src/components/content/patient-history/use-patient-history.tsx:169]: Maybe add comment here? Something like...

> The API returns responses sorted, so we can grab the first non-undefined response and that will be the latest.

**Review by WesleyKapow — APPROVED** [review]

**Pull request merged**